interface/main/messages/messages.php

```diff
@@ -23,614 +24,686 @@
-        $showall = "no";
-        $lnkvar="\"messages.php?show_all=yes&$activity_string_html\" name='See All' onclick=\"top.restoreSession()\">(".htmlspecialchars( xl('See All'), ENT_NOQUOTES).")";
-    }
-}
-?>
-<br>
-<table><tr><td><span class="title"> <?php echo htmlspecialchars(xl('Messages'), ENT_NOQUOTES); ?></span> <a class='more' href=<?php echo $lnkvar; ?></a></td></tr></table>
-<?php
-//show the activity links
-if (empty($task) || $task=="add" || $task=="delete") { ?>
-    <?php if ($active == "all") { ?>
-    <span><?php echo xlt('Show All'); ?></span>
-    <?php } else { ?>
-    <a href="messages.php" class="link" onclick="top.restoreSession()"><span><?php echo xlt('Show All'); ?></span></a>
-    <?php } ?>
-  |
-    <?php if ($active == '1') { ?>
-    <span><?php echo xlt('Show Active'); ?></span>
-    <?php } else { ?>
-    <a href="messages.php?form_active=1" class="link" onclick="top.restoreSession()"><span><?php echo xlt('Show Active'); ?></span></a>
-    <?php } ?>
-  |
-    <?php if ($active == '0') { ?>
-    <span><?php echo xlt('Show Inactive'); ?></span>
-    <?php } else { ?>
-    <a href="messages.php?form_inactive=1" class="link" onclick="top.restoreSession()"><span><?php echo xlt('Show Inactive'); ?></span></a>
-    <?php } ?>
-<?php } ?>
-
-<?php
-switch($task) {
-    case "add" :
-    {
-        // Add a new message for a specific patient; the message is documented in Patient Notes.
-        // Add a new message; it's treated as a new note in Patient Notes.
-        $note = $_POST['note'];
-        $noteid = $_POST['noteid'];
-        $form_note_type = $_POST['form_note_type'];
-        $form_message_status = $_POST['form_message_status'];
-        $reply_to = $_POST['reply_to'];
-        $assigned_to_list = explode(';', $_POST['assigned_to']);
-        foreach($assigned_to_list as $assigned_to){
-            if ($noteid && $assigned_to != '-patient-') {
-                updatePnote($noteid, $note, $form_note_type, $assigned_to, $form_message_status);
-                $noteid = '';
-            }
-            else {
-                if($noteid && $assigned_to == '-patient-'){
-                    // When $assigned_to == '-patient-' we don't update the current note, but
-                    // instead create a new one with the current note's body prepended and
-                    // attributed to the patient.  This seems to be all for the patient portal.
-                    $row = getPnoteById($noteid);
-                    if (! $row) die("getPnoteById() did not find id '".text($noteid)."'");
-                    $pres = sqlQuery("SELECT lname, fname " .
-                    "FROM patient_data WHERE pid = ?", array($reply_to) );
-                    $patientname = $pres['lname'] . ", " . $pres['fname'];
-                    $note .= "\n\n$patientname on ".$row['date']." wrote:\n\n";
-                    $note .= $row['body'];
+require_once("$srcdir/MedEx/API.php");
+
+$MedEx = new MedExApi\MedEx('MedExBank.com');
```

Comment:
See above comment about importing just `MedEx` which means this would work. Also, suggest refactoring `$MedEx` to `$medex` (Not sure if that's strictly PSR but we should aim for lowercase variables)